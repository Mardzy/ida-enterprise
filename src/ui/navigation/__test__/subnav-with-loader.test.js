import { fromJS } from "immutable";
import { Menu } from "semantic-ui-react";

import SubmenuWithLoader from "../subnav-with-loader";

describe("The submenu with loader", () => {

  it("displays an inline loader", () => {
    const wrapper = mount(<SubmenuWithLoader header="Some header" isLoading/>);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders a list of items", () => {
    const wrapper = mount(<SubmenuWithLoader header="Some header"
                                             items={fromJS([1,2,3])}
                                             renderItem={item => <Menu.Item key={item}>{item}</Menu.Item>}/>);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

});