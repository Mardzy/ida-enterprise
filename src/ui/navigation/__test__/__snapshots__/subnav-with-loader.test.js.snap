// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The submenu with loader displays an inline loader 1`] = `
<SubmenuWithLoader
  header="Some header"
  isLoading={true}
  items={Immutable.List []}
  renderItem={[Function]}
>
  <MenuItem>
    <div
      className="item"
      onClick={[Function]}
    >
      <MenuHeader>
        <div
          className="header"
        >
          Some header
           
          <InlineDots
            amount={3}
          >
            <styled.span
              key="0"
            >
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bdVaJa",
                      "isStatic": true,
                      "lastClassName": "gQgMD",
                      "rules": Array [
                        "
  animation: 1s ",
                        Keyframes {
                          "id": "sc-keyframes-cFFqjv",
                          "inject": [Function],
                          "name": "cFFqjv",
                          "rules": Array [
                            "@-webkit-keyframes cFFqjv{50%{color:transparent;}}",
                            "@keyframes cFFqjv{50%{color:transparent;}}",
                          ],
                          "toString": [Function],
                        },
                        " infinite;
  &:nth-child(1) {
    animation-delay: 0ms;
  }
  &:nth-child(2) {
    animation-delay: 250ms;
  }
  &:nth-child(3) {
    animation-delay: 500ms;
  }
",
                      ],
                    },
                    "displayName": "styled.span",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bdVaJa",
                    "target": "span",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <span
                  className="sc-bdVaJa gQgMD"
                >
                  .
                </span>
              </StyledComponent>
            </styled.span>
            <styled.span
              key="1"
            >
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bdVaJa",
                      "isStatic": true,
                      "lastClassName": "gQgMD",
                      "rules": Array [
                        "
  animation: 1s ",
                        Keyframes {
                          "id": "sc-keyframes-cFFqjv",
                          "inject": [Function],
                          "name": "cFFqjv",
                          "rules": Array [
                            "@-webkit-keyframes cFFqjv{50%{color:transparent;}}",
                            "@keyframes cFFqjv{50%{color:transparent;}}",
                          ],
                          "toString": [Function],
                        },
                        " infinite;
  &:nth-child(1) {
    animation-delay: 0ms;
  }
  &:nth-child(2) {
    animation-delay: 250ms;
  }
  &:nth-child(3) {
    animation-delay: 500ms;
  }
",
                      ],
                    },
                    "displayName": "styled.span",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bdVaJa",
                    "target": "span",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <span
                  className="sc-bdVaJa gQgMD"
                >
                  .
                </span>
              </StyledComponent>
            </styled.span>
            <styled.span
              key="2"
            >
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bdVaJa",
                      "isStatic": true,
                      "lastClassName": "gQgMD",
                      "rules": Array [
                        "
  animation: 1s ",
                        Keyframes {
                          "id": "sc-keyframes-cFFqjv",
                          "inject": [Function],
                          "name": "cFFqjv",
                          "rules": Array [
                            "@-webkit-keyframes cFFqjv{50%{color:transparent;}}",
                            "@keyframes cFFqjv{50%{color:transparent;}}",
                          ],
                          "toString": [Function],
                        },
                        " infinite;
  &:nth-child(1) {
    animation-delay: 0ms;
  }
  &:nth-child(2) {
    animation-delay: 250ms;
  }
  &:nth-child(3) {
    animation-delay: 500ms;
  }
",
                      ],
                    },
                    "displayName": "styled.span",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bdVaJa",
                    "target": "span",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <span
                  className="sc-bdVaJa gQgMD"
                >
                  .
                </span>
              </StyledComponent>
            </styled.span>
          </InlineDots>
        </div>
      </MenuHeader>
      <MenuMenu>
        <div
          className="menu"
        >
          <Component />
        </div>
      </MenuMenu>
    </div>
  </MenuItem>
</SubmenuWithLoader>
`;

exports[`The submenu with loader renders a list of items 1`] = `
<SubmenuWithLoader
  header="Some header"
  items={
    Immutable.List [
      1,
      2,
      3,
    ]
  }
  renderItem={[Function]}
>
  <MenuItem>
    <div
      className="item"
      onClick={[Function]}
    >
      <MenuHeader>
        <div
          className="header"
        >
          Some header
           
        </div>
      </MenuHeader>
      <MenuMenu>
        <div
          className="menu"
        >
          <MenuItem
            key="1"
          >
            <div
              className="item"
              onClick={[Function]}
            >
              1
            </div>
          </MenuItem>
          <MenuItem
            key="2"
          >
            <div
              className="item"
              onClick={[Function]}
            >
              2
            </div>
          </MenuItem>
          <MenuItem
            key="3"
          >
            <div
              className="item"
              onClick={[Function]}
            >
              3
            </div>
          </MenuItem>
        </div>
      </MenuMenu>
    </div>
  </MenuItem>
</SubmenuWithLoader>
`;
